src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -237,7 +237,9 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // not function properly.
             Href = output.Attributes[HrefAttributeName]?.Value as string;
 
-            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails);
+            var determinePartiallyMatchedAttributes = Logger.IsEnabled(
+                ModeMatchResultLoggerExtensions.PartiallyMatchedAttributesLogLevel);
+            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails, determinePartiallyMatchedAttributes);
             Logger.TagHelperModeMatchResult(modeResult, context.UniqueId, ViewContext.View.Path, this);
```

Comment:
Is this really important to have now that we've finished building these taghelpers? It would be nice to remove all of the code complexity if it's not needed.